remotes/workqueue.go

```diff
@@ -0,0 +1,83 @@
+package remotes
+
+import (
+	"context"
+
+	"golang.org/x/sync/errgroup"
+)
+
+type workItem struct {
+	do   func(ctx context.Context) error
+	done chan struct{}
+	deps []chan struct{}
+}
+
+func newWorkItem(do func(ctx context.Context) error, deps ...chan struct{}) *workItem {
+	return &workItem{
+		do:   do,
+		done: make(chan struct{}),
+		deps: deps,
+	}
+}
+
+func (wi *workItem) process(ctx context.Context) error {
+	defer close(wi.done)
+	for _, d := range wi.deps {
+		select {
+		case <-ctx.Done():
+			return nil
+		case <-d:
+		}
+	}
+	select {
+	case <-ctx.Done():
+		return nil
+	default:
+		return wi.do(ctx)
+	}
+}
+
+type workQueue struct {
+	workGroup *errgroup.Group
+	todoList  chan *workItem
+	ctx       context.Context
+}
+
+//nolint: unparam
```

Comment:
Is this because you are trying to conform to some other `interface`? Which param does it think is unneeded (and why is it wrong)?

Reply:
No, it is that the function is private, and is always called with the same constant parameter values.